### ISSUE TITLE
Error when type of dynamic route parameter is not supplied

### ISSUE THREAD

**Issue by blairj09** [trestletech/plumber#348]
Consider the following:
```r
library(plumber)

#* @param msg message to return to client
#* @get /msg/<msg>
function(msg = "") {
  list(msg = msg)
}
```

When this file is "plumbed", the following error occurs: 
```
Error in if (type == "bool" || type == "logical") { : 
  missing value where TRUE/FALSE needed
```
If the endpoint is changed to provide a type for the parameter, then it runs successfully. Both of the following work as expected.

```r
#* @param msg:character message to return to client
#* @get /msg/<msg>
function(msg = "") {
  list(msg = msg)
}
```

```r
#* @param msg message to return to client
#* @get /msg/<msg:chr>
function(msg = "") {
  list(msg = msg)
}
```

<details>
<summary>Session Info</summary>

``` r
devtools::session_info(pkgs = "plumber")
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.5.1 (2018-07-02)
#>  os       macOS  10.14.1              
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Denver              
#>  date     2018-11-19                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package  * version  date       lib source        
#>  BH         1.66.0-1 2018-02-13 [1] CRAN (R 3.5.0)
#>  crayon     1.3.4    2017-09-16 [1] CRAN (R 3.5.0)
#>  httpuv     1.4.5    2018-07-19 [1] CRAN (R 3.5.0)
#>  jsonlite   1.5      2017-06-01 [1] CRAN (R 3.5.0)
#>  later      0.7.5    2018-09-18 [1] CRAN (R 3.5.1)
#>  magrittr   1.5      2014-11-22 [1] CRAN (R 3.5.0)
#>  plumber    0.4.6    2018-06-05 [1] CRAN (R 3.5.0)
#>  promises   1.0.1    2018-04-13 [1] CRAN (R 3.5.0)
#>  R6         2.3.0    2018-10-04 [1] CRAN (R 3.5.1)
#>  Rcpp       1.0.0    2018-11-07 [1] CRAN (R 3.5.0)
#>  rlang      0.3.0.1  2018-10-25 [1] CRAN (R 3.5.0)
#>  stringi    1.2.4    2018-07-20 [1] CRAN (R 3.5.0)
#> 
#> [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
```

<sup>Created on 2018-11-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
</details>

**Comment by byzheng** [comment]
I have the same problem and the swagger doesn't support the dynamic parameters. See the curl command below. The {msg} don't replace with msg value.

![image](https://user-images.githubusercontent.com/1657381/48819238-b4457f00-ed9a-11e8-8363-877198301faa.png)

**Comment by schloerke** [comment]
Closed in #388
